recipes/javagotchas/descriptions/DonotexposeinternalCollection.html

```diff
@@ -0,0 +1,70 @@
+<h4>Abstract</h4>
+<p>Class methods should return immutable copies of private member variables of type <code>java.util.Collection</code> to prevent external changes to the state of the object. This is the encapsulation principle of OOP.</p>
+
+<h4>Description</h4>
+<p>Returning an instance's private field of type <code>java.util.Collection</code> allows external manipulation of the
+    internal state of an instance of the class because the collections are mutable. This can lead to unexpected program
+    behavior when external classes manipulate data in the collection, especially in multi-threaded situations.</p>
+<p>Class methods should return immutable copies of private member variables of type <code>java.util.Collection</code>
+</p>
+<table>
+    <tr>
+        <th>Before</th>
+        <th>After</th>
+    </tr>
+    <tr>
+        <td><pre>public class ListExample {
+    private List&#x3C;String&#x3E; myList;
+
+    public List&#x3C;String&#x3E; getMyList() {
+        return myList;
+    }
+}
+</pre>
+        </td>
+        <td><b>After applying Quickfix - Return an unmodifiable copy of the Collection</b>
+
+<pre>public class ListExample {
+    private List&#x3C;String&#x3E; myList;
+
+    public List&#x3C;String&#x3E; getMyList() {
+        return (List&#x3C;String&#x3E;)java.util.Collections.unmodifiableCollection(myList);
+    }
+}
+</pre>
+<b>After applying Quickfix - return an unmodifiable List</b>
+
+<pre>public class ListExample {
+    private List&#x3C;String&#x3E; myList;
+
+    public List&#x3C;String&#x3E; getMyList() {
+        return java.util.Collections.unmodifiableList(myList);
+    }
+}
+</pre>
+        </td>
+    </tr>
+    <tr>
+        <td><pre>public class SetExample {
+    private Set&#x3C;String&#x3E; mySet;
+
+    public Set&#x3C;String&#x3E; getSet() {
+        return mySet;
+    }
+}
+</pre>
+        </td>
+        <td><b>After applying Quickfix - return an unmodifiable Set</b>
+<pre>public class SetExample {
+    private Set&#x3C;String&#x3E; mySet;
+
+    public Set&#x3C;String&#x3E; getSet() {
+        return java.util.Collections.unmodifiableSet(mySet);
+    }
+}
+</pre>
+        </td>
+    </tr>
+</table>
+
+<h4>Resources</h4>
```

Comment:
```suggestion
<h2>Resources</h2>
```

Please add some resources if they're relevant otherwise delete the resource tag